src/config/load.c

```diff
@@ -228,6 +229,38 @@ cfgLoadUpdateOption(void)
             strPtr(cfgOptionStr(cfgOptRepoS3Bucket)));
     }
 
+    // Check/update compress-type if compress is valid. There should be no references to the compress option outside this block.
+    if (cfgOptionValid(cfgOptCompress) && cfgOptionSource(cfgOptCompress) != cfgSourceDefault)
+    {
+        if (cfgOptionSource(cfgOptCompressType) != cfgSourceDefault)
+        {
+            LOG_WARN(
+                "'" CFGOPT_COMPRESS "' and '" CFGOPT_COMPRESS_TYPE "' options should not both be set\n"
+                "HINT: '" CFGOPT_COMPRESS_TYPE "' is preferred and '" CFGOPT_COMPRESS "' is deprecated.");
+        }
+
+        // Set compress-type to none. Eventually the compress option will be deprecated and removed so this reduces code churn
+        // when that happens.
+        if (!cfgOptionBool(cfgOptCompress) && cfgOptionSource(cfgOptCompressType) == cfgSourceDefault)
+            cfgOptionSet(cfgOptCompressType, cfgSourceParam, VARSTR(compressTypeStr(compressTypeNone)));
+
+        // Now invalidate compress so it can't be used and won't be passed to child processes
+        cfgOptionValidSet(cfgOptCompress, false);
```

Comment:
Hmmm, yeah.  Looks like we need to move `cfgOptionSource(cfgOptCompress) != cfgSourceDefault` down as an if around lines 235-245.

The current way works fine since the option will never be passed to a child process when it is default, but better to make it invalid in all cases.